x/pki/handler_test.go

```diff
@@ -335,10 +335,11 @@ func TestHandler_ProposeAddX509RootCert_ForNocCertificate(t *testing.T) {
 
 	nocRootCertificate := rootCertificate(vendorAccAddress)
```

Comment:
I think we should have it already covered, but let's double check that we have tests for every index update for every operation.
Either create a dedicated test for it, or create a doc with the mapping of the test case to a particular existing unit test in code.

Something like a matrix: Action + Indices

**Actions:**
- Add DA Root
- Add DA Intermediate
- Revoke DA Root
- Revoke DA Intermediate
- Remove DA Root
- Remove DA Intermediate
- Add NOC Root
- Add NOC Intermediate
- Revoke NOC Root
- Revoke NOC Intermediate
- Remove NOC Root
- Remove NOC Intermediate

**Corresponding indices**
- AllCerts:
  - by Subject+SKID
  - by Subject
- DA:
  - by Subject+SKID
  - by Subject;
  - by SKID
- NOC:
  - by Subject+SKID
  - by Subject;
  - by SKID
  - by VID
  - by VID+SKID

Reply:
I refactored part of unit message handler unit tests a lot to use helpers doing check across all indexes, but we need to spend more time to   refactor all the tests.